DataSetExplorer/Program.cs

```diff
@@ -27,8 +33,91 @@ class Program
         static void Main(string[] args)
```

Comment:
This class currently owns many use cases that should be exported to a few services that can perform the coordination for each use case. Not something to solve as part of this PR, but definitely one of the next steps.